connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -586,24 +610,50 @@ object SparkSession extends Logging {
   }
 
   class Builder() extends Logging {
-    private var _client: SparkConnectClient = _
+    private val builder = SparkConnectClient.builder()
+    private var client: SparkConnectClient = _
 
     def remote(connectionString: String): Builder = {
-      client(SparkConnectClient.builder().connectionString(connectionString).build())
+      builder.connectionString(connectionString)
       this
     }
 
     private[sql] def client(client: SparkConnectClient): Builder = {
-      _client = client
+      this.client = client
       this
     }
 
-    def build(): SparkSession = {
-      if (_client == null) {
-        _client = SparkConnectClient.builder().build()
+    private def tryCreateSessionFromClient(): Option[SparkSession] = {
+      if (client != null) {
+        Option(new SparkSession(client, cleaner, planIdGenerator))
+      } else {
+        None
       }
-      new SparkSession(_client, cleaner, planIdGenerator)
     }
+
+    /**
+     * Build the [[SparkSession]].
+     *
+     * This will try to reuse an existing session if has the same configuration.
+     */
+    def build(): SparkSession = {
```

Comment:
While this might be compatible behavior with the SparkSession, can we please create a `create` method that instantiates guaranteed a new session?

See https://github.com/apache/spark/pull/41013

Reply:
It is below this method right?

Reply:
Hm, `SparkSession` doesn't have `build`. Shall we at least deprecate?